src/platforms/php/guides/symfony/index.mdx

```diff
@@ -37,7 +37,15 @@ Add your DSN to `config/packages/sentry.yaml`:
 
 ```yaml {filename:config/packages/sentry.yaml}
 sentry:
-    dsn: "___PUBLIC_DSN___"
+    dsn: "%env(SENTRY_DSN)%"
+```
+
+And in your `.env` file:
+
+```env {filename:.env}
+###> sentry/sentry-symfony ###
```

Comment:
If you set the DSN in this file it will be loaded for all environments, which is wrong for obvious reasons. We should instead tell users that they have to set this environment variable in their own shell

Reply:
I would go with the uncommited `.env.local` file here which is used by the devs. The real ENV is only when you use a docker setup or some cloud hosting where you are able to set a global ENV.

Reply:
That's also a solution of course, but it must be clear that having a DSN set locally will enable the bundle, and setting it in that file won't work for production 😉

Reply:
So I would go witht something like this here then:

~~~
The `SENTRY_DSN` need to be defined as ENVIRONMENT variable or you can be set it in your uncommitet `.env.local` file:


```env {filename:.env.local}
SENTRY_DSN="___PUBLIC_DSN___"
```
~~~

Reply:
The point is: once you set it in `.env.local`, it will work only for the local environment (at least according to the canonical meaning of that file), so I see little value in suggesting people to set it like that because you usually use Sentry just for the production environment. Instead, if it's set in that file, it will again be enabled for all environments on your own machine, which makes no sense to me

Reply:
Sure thats clear, but think as Symfony dev you should know that as you also need to set `APP_ENV=prod` and your database connection as env variable or in the `.env.local` file. The thing is that most still use some kind of shared servers for hosting there projects and so there the `.env.local` file is more common as you can not set a global env variable. Also devs are more common friendly with the `.env.local` file in symfony then how to set a ENV variable in there webserver, sound strange but my experience was it that most still just use `.env.local` file to configure APP_ENV, DATABASE_URL, ... in symfony.

Reply:
Hmm this is one of the suggestions coming from #3233. Prior to that we were setting the DSN (___PUBLIC_DSN___) straight in the config file.

What ya'll think is best here then?

Reply:
>Prior to that we were setting the DSN (`___PUBLIC_DSN___`) straight in the config file

Also that was incorrect then. I think that setting things using environment variables is now considered a best practice in the Symfony world since a lot of years, so we should definitely use them

> The thing is that most still use some kind of shared servers for hosting there projects and so there the .env.local file is more common as you can not set a global env variable

I don't have any data to say this is not true, although I strongly believe that Symfony doesn't even run on a shared hosting due to performance issues :sweat_smile: Anyway, `.env.prod.local` should be the right file to use so that it's still a local file not committed to the repo and it works only in production. WDYT?

Reply:
Considering that this `.env` solution is essentially the same as setting a DSN directly in config as we document today (see below), I'll err on the side of iterating quick and merging this in like this. Please send a new PR if you think this is confusing and there's a better way.


---

https://docs.sentry.io/platforms/php/guides/symfony/#configure

![image](https://user-images.githubusercontent.com/88819/112882301-116d4000-90cd-11eb-939a-94e7ea57ca09.png)

Reply:
We don't go to this level of detail in any framework docs that I'm aware of, and where the same dev/staging/prod considerations exist.

For example:
- https://docs.sentry.io/platforms/python/guides/django/#configure
- https://docs.sentry.io/platforms/python/guides/flask/#configure
- https://docs.sentry.io/platforms/java/guides/spring-boot/#configure

In the end of the day, developers and teams know better how their per-environment configuration is supposed to be handled, and there is no one right way to do it.

That said, I wouldn't oppose any contributions that improve the Symfony docs to include more specific advice, it is just that I'm not the best person to drive that effort.